### PR TITLE
Add Dockerfile and Docker Compose for local development

### LINKED ISSUE #5: Add Docker Compose for local development
## Tasks
- [ ] docker-compose.yml with MongoDB, Kafka (Confluent), Zookeeper, cartservice
- [ ] Dockerfile (multi-stage build, same pattern as productservice)
- [ ] .dockerignore

## Services
- MongoDB 7 on port 27017
- Zookeeper + Kafka (Confluent 7.6.0) on port 9092
- Cart service on port 8082
- Shared vibevault-network for inter-service communication

### PR DESCRIPTION
**Pull request by spa-raj**
## Summary
- Multi-stage Dockerfile (JDK builder + JRE runtime), non-root user, health check, `exec` entrypoint for signal handling
- Docker Compose with MongoDB 7, Kafka in KRaft mode (Confluent 7.8.7, no Zookeeper)
- Cart service on port 8082, shared vibevault-network for inter-service communication
- Cart API test script with 13 test sections including Kafka event verification and cart isolation

## Services
| Service | Image | Port | Mode |
|---------|-------|------|------|
| MongoDB | mongo:7 | 27017 | Standalone |
| Kafka | confluentinc/cp-kafka:7.8.7 | 9092 | KRaft (no Zookeeper) |
| Cart Service | cartservice (built) | 8082 | - |

## Test script (scripts/test-cart-apis.sh)
13 sections covering:
1-3: Health checks, OAuth2 token, test product setup
4-6: Cart CRUD (empty cart, add items, get cart)
7-8: Update quantity, remove item
9-10: Checkout, clear cart
11: Edge cases (validation, auth, empty cart checkout)
12: Kafka event verification (ITEM_ADDED, ITEM_UPDATED, ITEM_REMOVED, CHECKOUT_INITIATED, CART_CLEARED)
13: Cart isolation (multi-user — verifies users can only access their own cart)

## Local dev setup
docker network create vibevault-network
docker compose up -d
./scripts/test-cart-apis.sh

Requires userservice (8081) and productservice (8080) running on vibevault-network.

Fixes #5

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      - "9092:9092"
+    environment:
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:29092,PLAINTEXT_HOST://localhost:9092
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_AUTO_CREATE_TOPICS_ENABLE: "true"
+    healthcheck:
+      test: ["CMD", "kafka-topics", "--bootstrap-server", "localhost:9092", "--list"]
```
[the remark this answers is not in the document]
Comment (spa-raj) [reply docker-compose.yml]: Not fixing. Both listeners (9092 and 29092) are served by the same broker process — if one is up, both are up. The healthcheck on 9092 is sufficient.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# Health check
+HEALTHCHECK --interval=30s --timeout=10s --start-period=60s --retries=3 \
+    CMD wget --no-verbose --tries=1 --spider http://localhost:8082/actuator/health || exit 1
+
+# JVM options for container environment
+ENV JAVA_OPTS="-XX:+UseContainerSupport -XX:MaxRAMPercentage=75.0"
+
+# Entrypoint
+ENTRYPOINT ["sh", "-c", "java $JAVA_OPTS -jar app.jar"]
```
[the remark this answers is not in the document]
Comment (spa-raj) [reply Dockerfile]: Fixed. Added `exec` so Java becomes PID 1 for proper signal handling.

File: src/main/java/com/vibevault/cartservice/configurations/KafkaProducerConfig.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+public class KafkaProducerConfig {
+
+    @Bean
+    public ProducerFactory<String, CartEvent> producerFactory(
+            @Value("${spring.kafka.bootstrap-servers}") String bootstrapServers) {
+        Map<String, Object> config = new HashMap<>();
+        config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        config.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSerializer.class);
+        return new DefaultKafkaProducerFactory<>(config);
```
[the remark this answers is not in the document]
Comment (spa-raj) [reply src/main/java/com/vibevault/cartservice/configurations/KafkaProducerConfig.java]: Fixed. Switched to `KafkaProperties.buildProducerProperties()` with `spring-boot-starter-kafka`.
